src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java

```diff
@@ -733,9 +777,15 @@ public final Response cancelBooking(@Context final HttpServletRequest request,
             }
 
             // if the user id is null then it means they are changing their own booking.
-            if (userId != null && !isUserAnAdminOrEventManager(userManager, request) ) {
-                return new SegueErrorResponse(Status.FORBIDDEN, "You must be an admin user to change another user's booking.")
-                    .toResponse();
+            if (userId != null) {
+                if(isUserAbleToManageEvents(userManager, request)) {
+                    return new SegueErrorResponse(Status.FORBIDDEN, "You must be an admin user to change another user's booking.")
+                            .toResponse();
+                }
+                if (userLoggedIn.getRole() == Role.EVENT_LEADER && !userAssociationManager.hasPermission(userLoggedIn, userOwningBooking)) {
```

Comment:
Not sure if we have made a particular style decision on this but I would always try and use .equals here even for the enum. If you choose to do this I would invert it so that it is Role.EVENT_LEADER.equals(x) to avoid null pointer exceptions. It is a preference thing though so happy if you want to ignore this one.